proc_IR_noecho.py

```diff
@@ -31,27 +34,45 @@
             expno=which_exp, 
             postproc=postproc, lookup=postproc_dict, 
             dimname='indirect')
-    #{{{convolution and zeroth order phase correction
-    fl.next('select coherence pathway and convolve')
-    s = s['ph2',0]['ph1',-1]
-    s.convolve('t2',5)
+    s.ift('t2')
+    rough_center = abs(s).convolve('t2',0.01).mean_all_but('t2').argmax('t2').item()
+    s.setaxis(t2-rough_center)
+    fl.next('rough centered')
     fl.image(s)
+    #{{{convolution and zeroth order phase correction
+    s = s['t2':(0,None)]['ph2',0]['ph1',-1]
     s.reorder('t2')
     ph0 = zeroth_order_ph(s['t2':0],fl=None)
     ph0 /= abs(ph0)
     s /= ph0
+    fl.next('zeroth order phase correction')
+    fl.image(s)
     #}}}
     #{{{visualize phased spectra
     for j in range(ndshape(s)['indirect']):
         fl.next('Plotting phased spectra')
-        fl.plot(s['indirect',j]['t2':(-150,150)],
+        fl.plot(s['indirect',j]['t2':(0,None)],
             alpha=0.5,
             label='vd=%g'%s.getaxis('indirect')[j])
     #}}}
-    #{{{exponential curve
+    #{{{exponential curve with fit
+    s.ft('t2')
```

Comment:
works on a524dcad0c2c33b3e7c93a81bf405d14e782928f of pyspecdata but not on current guess_function branch; produces error:
`Traceback (most recent call last):
  File "proc_IR_noecho.py", line 59, in <module>
    s.ft('t2')
  File "c:\users\alex guinness\gitstuff\pyspecdata\pyspecdata\fourier\ft.py", line 216, in ft
    " on??").format(du,p2_pre_discrepancy)
AssertionError: I expect the discrepancy to be smaller than du (0.00), but it's 0.00 -- what's going on??`
when s.ft('t2') is commented out it produces this error:
`Traceback (most recent call last):
  File "proc_IR_noecho.py", line 60, in <module>
    rec_curve = s['t2':(-150,150)].sum('t2')
  File "c:\users\alex guinness\gitstuff\pyspecdata\pyspecdata\core.py", line 6294, in __getitem__
    slicedict,axesdict,errordict,unitsdict = self._parse_slices(args)
  File "c:\users\alex guinness\gitstuff\pyspecdata\pyspecdata\core.py", line 6486, in _parse_slices
    logger.debug(strm("i found",thisaxis[temp_high]))
IndexError: index 1971 is out of bounds for axis 0 with size 1971`

Reply:
I believe this is the same error as previous + not  a bad fix

Reply:
runs without error now (though the plots look a little weird to me)

Reply:
So I'm hoping the adjustments I made don't roll us back. I implemented portions from proc_IR.py (might be easier in future to integrate another second level function for these two). There is an issue with nnls inside BRD stuff in pyspecdata. Before messing around I wanted to see if you knew an easy fix before delving into that rabbit hole. The error is:
'''Traceback (most recent call last):                                                                                        File "proc_IR_noecho.py", line 134, in <module>                                                                           soln = s.real.nnls('indirect',T1, lambda x,y: 1.0-2.*exp(-x/y),l=this_l)                                              File "c:\users\alex guinness\gitstuff\pyspecdata\pyspecdata\core.py", line 4842, in nnls                                  residual_nddata.data[:] = residual[:]                                                                               ValueError: could not broadcast input array from shape (15856) into shape (4,2,1982)   '''

Reply:
so, it's easy to not get this error -- you would change the rhs to `residual_nddata.data.ravel()[:]` (here I'm assuming that the thing on the right is the thing that's 4x2x1982)  **but** it begs the question: why is this thing that shape -- that doesn't make much sense.  Also, why wasn't this happening before?  What has changed between "then" and "now"?  I don't think that nnls on master has changed recently (Alec has been doing some work, but that's not merged yet)

Reply:
I never checked in on this issue but it no longer is occurring. I have since merged the master into guess function branch on pyspecdata and all seems well